### ISSUE TITLE
Our 100% coverage is intermittent

### ISSUE THREAD

**Issue by daleharvey** [pouchdb/pouchdb#4767]

**Comment by nolanlawson** [comment]
I'm guessing this is because some code paths in `sync.js` or `replicate.js` are only occasionally hit, depending on the offline/retry tests.

**Comment by daleharvey** [comment]
Yeh sorry meant to follow this up, its exactly that, one of the sync paused rarely gets hit, will put up a patch for it shortly

**Comment by daleharvey** [comment]
Hrm, so https://github.com/pouchdb/pouchdb/commit/443cd66992337093ec963a61d1d3d333d63c6a4b has apparently regressed our coverage, confused how but looking into it, @nolanlawson if you have an idea would appreciate

**Comment by nolanlawson** [comment]
Ah, I know what's happening.

So the issue is that the unit tests can _only_ work with `npm run build-as-modular-es5`, because they need to reach "inside" the rollup bundle to test things that normally aren't exported.

The problem comes when we then want to test for coverage, because "normally" we'd want to test on the bundle, but some functions are only covered by the unit tests, so we need to test on the modular ES5 instead.

When you added your change in 443cd66, it caused Travis to build once normally (on `npm install`), then once for modular ES5 (in `npm run test-unit`), then _once again_ normally, meaning that the `lib` directory is a mix of modular and non-modular code. The non-modular code is tested by the non-unit tests, whereas the modular code is tested by the unit tests.

TLDR: The ~97% we're now at represents what's covered by the non-unit tests.

**Comment by daleharvey** [comment]
Ok the only one left is the sync the gets not hit very very occasionally, gonna try and get a test for that part
